### PR TITLE
fix: fix argument and return value handling for get_by_id

### PR DESCRIPTION
**Pull request by jgadling**
Fix: #1379 

We broke the interface for `get_by_id` when improving the generated documentation for the client. This fixes the interface and adds a test.

### CODE REVIEW

**Review by andy-sweet — APPROVED** [review]
This fixes the issue for me and is probably the smallest possible change to fix, so approved. I would personally prefer that the template uses the exact same signature as the base class (i.e. `def get_by_id(cls, client, id)`), but maybe I'm not following the doc generation code carefully enough.

File: client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2 [hunk 1]
```diff
@@ -43,14 +43,14 @@ class {{ model.name }}(Model):
     find.__func__.__doc__ = Model.find.__func__.__doc__ + find.__func__.__doc__
 
     @classmethod
-    def get_by_id(cls, **kwargs):
+    def get_by_id(cls, *args, **kwargs):
```
Comment (uermel) [inline client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2:46]: Not a blocker, but is there any way we can retain the `client, id` signature here?

It's a bit concerning to me that in the process of improving the web docs we're making IDE-based code inspection worse.

File: client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2 [hunk 2]
```diff
@@ -43,14 +43,14 @@ class {{ model.name }}(Model):
...
         """
         Examples:
             Get an {{ model.name }} by ID:
             >>> {{ model.model_name_underscores }} = {{ model.name }}.get_by_id(client, 1)
```
Comment (uermel) [inline client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2:50]: Also not a blocker, but we should find a way to insert class-specific IDs here. The `id=1` will return `None` for almost every important model in the client, so these are not necessarily intuitive examples. I will create a ticket.

**Review by uermel — APPROVED** [review]
Thank you for the quick fix!

File: client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2 (hunk 2 above)
Comment (uermel) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2:50]: Captured here: https://github.com/chanzuckerberg/cryoet-data-portal/issues/1381

File: client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2 (hunk 1 above)
Comment (jgadling) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2]: Fair enough - I've updated this PR to make the method signatures more explicit. Mind taking another look?

Comment (uermel) [reply client/python/cryoet_data_portal/src/cryoet_data_portal/templates/Model.jinja2]: Yep, this works for me in PyCharm.

**Pull request merged**